File: Semestres/1.Febrero-Junio 2017/FADATulua/20Mayo/EjercicioClase.cpp
#include <iostream>
#include <string>
#include <queue>
#include <stack>
#include <map>
using namespace std;

int main(){
	//string texto = "([<{abc123abc}>])";
	string texto = "(abc[123)abc]";
	queue<char> colita;
	stack<char> pila;
	map<char,char> mapa;
	
	mapa['('] = ')';
	mapa['['] = ']';
	mapa['<'] = '>';
	mapa['{'] = '}';
	mapa[')'] = '(';
	mapa['}'] = '{';
	mapa['>'] = '<';
	mapa[']'] = '[';

	
	string brackets = "[({<>})]";
	
	for(int i=0; i<(int)texto.size(); i++){
		for(int j=0; j<(int)brackets.size(); j++){
			if(texto[i] == brackets[j]){
				colita.push(texto[i]);
				pila.push(texto[i]);
			}
		}
		
	}

	
	for(int i=0; i<(int)colita.size(); i++){
		if(mapa[colita.front()] != pila.top()){
			cout << "false" << endl;
			return 0;
		}
		colita.pop();
		pila.pop();
	}
	cout << "true" << endl;
}

File: Semestres/1.Febrero-Junio 2017/FADATulua/20Mayo/EjercicioClase.java
import java.util.*;
import java.util.LinkedList;

public class EjercicioClase{
	
	public static void main(String args[]){
		
		String texto = "([<{abc123abc}>])";
		
		Queue<Character>  colita = new LinkedList();
		Stack<Character> pila = new Stack<Character>();	
		Map<Character, Character> mapa = new HashMap<Character, Character>();

	
		mapa.put('(',')');
		mapa.put('[',']');
		mapa.put('<', '>');
		mapa.put('{', '}');
		mapa.put(')', '(');
		mapa.put('}','{');
		mapa.put('>', '<');
		mapa.put(']', '[');

	
		String brackets = "[({<>})]";
		
		for(int i=0; i<(int)texto.length(); i++){
			for(int j=0; j<(int)brackets.length(); j++){
				if(texto.charAt(i) == brackets.charAt(j)){
					colita.add(texto[i]);
					pila.push(texto[i]);
				}
			}
		
		}

	
		for(int i=0; i<(int)texto.length(); i++){
			if(mapa.get(colita.peek()) != pila.peek()){
				System.out.println("false");
				return;
			}
			colita.remove();
			pila.pop();
		}
		System.out.println("true");
				
	}
}
